### PR TITLE
FacilityInfo timezone support

### LINKED ISSUE #22739: Timestamps for BST are wrong in python
The error was presented on the [forum](http://forum.mantidproject.org/t/sample-environment-log-timestamps-wrong/409). The fix should include a unit test.

It turns out that `numpy` already has the functionality to do the right thing using the `datetime_as_string`.
```python
print 'US    ', np.datetime_as_string(t[0], timezone=pytz.timezone('US/Eastern'))
print 'London', np.datetime_as_string(t[0], timezone=pytz.timezone('Europe/London'))
print 'Paris ', np.datetime_as_string(t[0], timezone=pytz.timezone('Europe/Copenhagen'))
```
The PR will include an additional field in the `Facilities.xml` to contain the correct timezone (which numpy uses with its knowledge of daylight savings). In C++ `FacilityInfo.timezone()` will return the string. In python `FacilityInfo.timezone()` will return the `pytz.timezone` result.

The PR will also include new documentation in [`DateAndTime`](http://docs.mantidproject.org/nightly/api/python/mantid/kernel/DateAndTime.html) explaining how to use this functionality.

### PR DESCRIPTION
**Pull request by peterfpeterson**
Add a new field on `FacilityInfo` which gives the timezone that the facility is in and update the documentation in `DateAndTime` to help users take advantage of this.

**To test:**

See what can be done based off of the new example in `DateAndTime`. `np.datetime64` can be printed with the timezone of where they were measured rather than system timezone.

Fixes #22739.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [x] Are the unit tests small and test the class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] Do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by soininen — CHANGES_REQUESTED** [review]
The timezone stuff seems to work as expected. Having more than one timezone for a facility aborts at startup of MantidPlot but the error message is clearly visible so I think that is OK. However, if no timezone is defined at all, the unit test will fail and the `pytz` magic doesn't work for empty strings. Does this make the timezone entry in `Facilities.xml` mandatory? If so, then the corresponding [Concepts page](http://docs.mantidproject.org/nightly/concepts/FacilitiesFile.html) should be updated accordingly.

**Comment by peterfpeterson** [comment]
Good catch. In principle it is optional, but I didn't think it was that hard to pick a nearby city that a facility shares timezone with. I'll update the concepts page with instructions on how to get a list of valid timezones out of pytz.

**Review by soininen — APPROVED** [review]

File: Framework/CurveFitting/src/Algorithms/ConvolutionFit.cpp [hunk 1]
```diff
@@ -255,7 +255,7 @@ ConvolutionFit<QENSFitSequential>::seeAlso() const {
 template <>
 const std::vector<std::string>
 ConvolutionFit<QENSFitSimultaneous>::seeAlso() const {
-  return {"QENSSimultaneousFit"};
+  return {"QENSFitSimultaneous"};
```
Comment (martyngigg) [inline Framework/CurveFitting/src/Algorithms/ConvolutionFit.cpp]: @brandonhewer This changes looks like the correct thing to do based on the class name but are there any implications for the MSG GUIs?

**Comment by martyngigg** [comment]
I think we need to update the [schema](https://github.com/mantidproject/mantid/blob/master/instrument/Schema/Facilities/1.0/FacilitiesSchema.xsd#L1) too?

File: Framework/CurveFitting/src/Algorithms/ConvolutionFit.cpp (hunk 1 above)
Comment (NickDraper) [reply Framework/CurveFitting/src/Algorithms/ConvolutionFit.cpp]: It's only the seeAlso string, I don't think they are used anywhere but in the documentation.

Comment (martyngigg) [reply Framework/CurveFitting/src/Algorithms/ConvolutionFit.cpp]: Good point. Thanks @NickDraper.

**Review by martyngigg — CHANGES_REQUESTED** [review]
Marking to request changes that the facilities schema is updated.

**Review by martyngigg — APPROVED** [review]
The `ValidateFacilitiesFile` test now passes with the schema update.

**Pull request merged**